### DISCUSSION TITLE
Octopus saving session optmization

### DISCUSSION THREAD

**Discussion by marcoaddario** [springfall2008/batpred#805] [Ideas]
Hi all,
I was looking today at the plan just after I got the message that I have joined the today's saving session.
This is a behavior I already noticed in the past but I then forgot to write about it (so, is not directly related to today's session or this specific predbat version).
the session today will last for 30 minutes. 
predbad knows how much I will be able to discharge in these 30 minutes. So my question is: wouldn't be more optimized to only charge for what I will be able to discharge? 
Probably there is a good reason behind this; but if not, do you think this is a space for an improvement?

<img width="584" alt="image" src="https://github.com/springfall2008/batpred/assets/3762730/00b55dce-72ee-4091-b3ac-9c3079749d75">

**Comment by gcoan** [comment]
Bit late in replying, and its all after the event now, but I think what you are asking for is already happening.

Predbat knows how much you can discharge as you say, and in the plan you can see that the SoC reduces from 96% to 69% after the discharge slot.

The charging beforehand should be calculated by predbat to be whatever you need to deliver the entire forecast plan in the most cost optimal way.   Although there is a long charge period, its actually only charging the SoC from 89% to 96%.

Would really need to see the rest of the plan through the night to see whether this pre-charge was necessary or not, but the basic logic is that its a full plan cost optimisation.
